### PR TITLE
API review changes

### PR DESCRIPTION
**Pull request by mikaelm12**
Changed the abstract class Rule to an the interface IRule.
Renamed enum RuleTermination to RuleResult and renamed values. 
@natemcmaster @Tratcher @davidfowl @glennc @Eilon

### CODE REVIEW

**Comment by natemcmaster** [comment]
Looks like you'll need to rebase against dev.

File: src/Microsoft.AspNetCore.Rewrite/Internal/ApacheModRewrite/ApacheModRewriteRule.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public ApacheModRewriteRule(UrlMatch initialMatch, IList<Condition> conditions,
             Actions = urlActions;
         }
 
-        public override void ApplyRule(RewriteContext context)
+        public void ApplyRule(RewriteContext context)
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/Internal/ApacheModRewrite/ApacheModRewriteRule.cs]: Let's make this `virtual` so users can override if they need.

File: src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/IISUrlRewriteRule.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ public class IISUrlRewriteRule : Rule
             Action = action;
         }
 
-        public override void ApplyRule(RewriteContext context)
+        public void ApplyRule(RewriteContext context)
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/IISUrlRewriteRule.cs]: `virtual`

File: src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/UrlRewriteFileParser.cs [hunk 3]
```diff
@@ -81,7 +81,7 @@ private void ParseRuleAttributes(XElement rule, UrlRewriteRuleBuilder builder)
             var action = rule.Element(RewriteTags.Action);
             if (action == null)
             {
-                ThrowUrlFormatException(rule, "Rule does not have an associated action attribute");
+                ThrowUrlFormatException(rule, "IRule does not have an associated action attribute");
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/UrlRewriteFileParser.cs]: Can stay as just "Rule".

File: src/Microsoft.AspNetCore.Rewrite/Internal/RewriteRule.cs [hunk 4]
```diff
@@ -31,7 +31,7 @@ public RewriteRule(string regex, string replacement, bool stopProcessing)
             StopProcessing = stopProcessing;
         }
 
-        public override void ApplyRule(RewriteContext context)
+        public void ApplyRule(RewriteContext context)
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/Internal/RewriteRule.cs]: `virtual`

File: src/Microsoft.AspNetCore.Rewrite/RuleResult.cs [hunk 5]
```diff
@@ -6,19 +6,19 @@ namespace Microsoft.AspNetCore.Rewrite
...
-    public enum RuleTermination
+    public enum RuleResult
     {
         /// <summary>
         /// Default value, continue applying rules.
         /// </summary>
-        Continue,
+        ContinueRules,
         ///<summary>
         /// Redirect occured, should send back new rewritten url.
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/RuleResult.cs]: Can you update this description to `The rule ended the request by providing a response.`

**Comment by troydai** [comment]
You may need to rebase to eliminate conflicts. But besides that it looks fine.

:shipit:

**Pull request merged**